gui/wxpython/iclass/digit.py

```diff
@@ -147,9 +149,15 @@ def CopyMap(self, name, tmp=False):
         poMapInfoNew = pointer(Map_info())
 
         if not tmp:
-            open_fn = Vect_open_new
+            if update:
+                open_fn = Vect_open_update
+            else:
+                open_fn = Vect_open_new
         else:
-            open_fn = Vect_open_tmp_new
+            if update:
+                open_fn = Vect_open_tmp_update
+            else:
+                open_fn = Vect_open_tmp_new
 
         is3D = bool(Vect_is_3d(self.poMapInfo))
         if open_fn(poMapInfoNew, name, is3D) == -1:
```

Comment:
Vect_open_tmp_update and Vect_open_tmp_new have different third argument (3d vs mapset), but here you pass is3D in both cases.

Reply:
Thanks. I've already fixed it.